prog/storage/migrate_spdk_vm_to_ubiblk.rb

```diff
@@ -0,0 +1,184 @@
+# frozen_string_literal: true
+
+class Prog::Storage::MigrateSpdkVmToUbiblk < Prog::Base
+  subject_is :vm
+
+  def self.assemble(vm_id)
+    unless (vm = Vm[vm_id])
+      fail "Vm does not exist"
+    end
+
+    unless vm.vm_storage_volumes.count == 1
+      fail "This prog only supports Vms with exactly one disk"
+    end
+
+    unless vm.vm_storage_volumes.first.use_bdev_ubi
+      fail "Vm is already using Ubiblk"
+    end
+
+    unless vm.vm_host.vhost_block_backends.find { |b| b.version == Config.vhost_block_backend_version }
+      fail "VmHost does not have the right vhost block backend installed"
+    end
+
+    Strand.create(
+      prog: "Storage::MigrateSpdkVmToUbiblk",
+      label: "stop_vm",
+      stack: [{
+        "subject_id" => vm_id
+      }]
+    )
+  end
+
+  label def stop_vm
+    vm.incr_stop
```

Comment:
Maybe also `register_deadline(nil, 60 * 60)` so we get a page and don't forget about this if the migration fails.